python/tvm/auto_scheduler/search_task.py

```diff
@@ -221,10 +221,6 @@ def __init__(
             target_host = Target(target_host)
 
         self.dag = compute_dag
```

Comment:
I don't think we can access the constructed object in this constructor? The point is both SearchTask and ConouteDAG have to be reconstructed when deserilization, we need to call their C++ constructors in `__setstate__`.

Reply:
@jcf94 the root cause is we have two ComputeDAG constructors in C++, one takes `compute` and another takes `schedule`. Since we can only register one `auto_scheduler.ComputeDAG` symbol to Python, I made a constructor dispatching as follows. Recall that this is to preserve the stage order if we already have a schedule and wish to use it to create a ComputeDAG.

```
      if (tensors) {
        return ComputeDAG(tensors.value());
      }
      ICHECK(sch) << "Both tensors and schedule are null";
      return ComputeDAG(sch.value());
```

It means we need both `tensors` (named `compute` on the Python side) and `sch` to call the ComputeDAG constructor from Python. If we really want to keep one, then we should keep `sch` because it has more information than `tensors`. Since `sch` is not stored in the C++ object so we cannot access it via FFI, the current workaround is having a Python-side specific attribute `self.sche`. However, this attribute will be wiped out when mapping to the C++ object, so the current solution is keeping a Python object.

Another solution would be keeping `sch` in the ComputeDAG C++ object. I made a new commit for this solution. Please comment which one you prefer.

Reply:
Oh, thanks! Yeah, I finally get yout point. Let's have a discusse in the meeting.